Sources/Amplitude/Amplitude.m

```diff
@@ -284,6 +272,49 @@ - (instancetype)initWithInstanceName:(NSString *)instanceName {
     return self;
 }
 
+- (void)migrateToFileStorage {
+     NSString *databaseDirectory = [AMPUtils platformDataDirectory];
+     NSString *databasePath = [databaseDirectory stringByAppendingPathComponent:@"com.amplitude.database"];
+    if (![self->_instanceName isEqualToString:kAMPDefaultInstance]) {
+        databasePath = [NSString stringWithFormat:@"%@_%@", databasePath, self.instanceName];
+    }
+    NSString *fileStoragePath =  [AMPStorage getAppStorageAmpDir:self.instanceName];
+
+    BOOL firstMigration = ![[NSFileManager defaultManager] fileExistsAtPath:fileStoragePath isDirectory: true];
+    BOOL needDBMigrate =  ([[NSFileManager defaultManager] fileExistsAtPath:databasePath] && firstMigration) ? true : false;
```

Comment:
So clarify the logic a bit, firstMigration is true when there is no new file storage, and that's also why getAppStorageAmpDir is changed to public. And needDBMigrate is checking if we have the database now?

Reply:
Yes, you are right.